### PR TITLE
Fixes #32672 - Add helpers for operating system's version facts

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

**Comment by upadhyeammit** [comment]
@evgeni request to review thank you!

File: lib/foreman_maintain/concerns/system_helpers.rb [hunk 1]
```diff
@@ -192,6 +192,31 @@ def check_version(name)
           yield current_version
         end
       end
+
+      def all_system_facts
+        facter = ForemanMaintain::Utils::Facter.path
+        @all_system_facts ||= JSON.parse(execute("#{facter} -j"))
```
Comment (evgeni) [inline lib/foreman_maintain/concerns/system_helpers.rb:198]: Given we only look at `os` facts, isn't it cheaper to call `facter -j os` and call this `os_facts` or something?

Comment (evgeni) [reply lib/foreman_maintain/concerns/system_helpers.rb:198]:
```
# time facter -j os
…
real	0m0.264s
user	0m0.216s
sys	0m0.048s

# time facter -j
…
real	0m0.392s
user	0m0.260s
sys	0m0.123s
```

**Comment by evgeni** [comment]
One minor comment, but otherwise makes total sense!

File: lib/foreman_maintain/concerns/system_helpers.rb (hunk 1 above)
Comment (upadhyeammit) [reply lib/foreman_maintain/concerns/system_helpers.rb]: Resolved. I thought maybe all facts can help in future however I agree that if needed we can change it later.

**Review by evgeni — APPROVED** [review]

**Pull request merged**